### DISCUSSION TITLE
Compiling NFFT instructions (git clone fatal error, step 2)

### DISCUSSION THREAD

**Discussion by zmichels** [mtex-toolbox/mtex#1140] [Tips and Tricks]
Hi,

I am following the install instructions for NFFT on a mac computer.

I kept getting the following error at Step 2: 
Get nfft source from github
git clone https://github.com/NFFT/nfft/tree/develop
./bootstrap.sh

Error is:
fatal: repository 'https://github.com/NFFT/nfft/tree/develop/' not found

after some web searches, it seems maybe the trouble is that the specific URL is "private" or something, and despite "solutions" I found online, none worked.

I finally removed the end of the URL such that the terminal command was:
git clone https://github.com/NFFT/nfft/

... and that completed fine.

Still struggling to get the remainder of the install procedure complete, but I'll report back if anything works eventually

**Comment by Mary-alix** [comment]
Hello, 

I use mtex 5.6.1, and Matlab 2020b. 
The work-around that zmichels suggested also gets me passed step 2. Now I am stuck.

For step 3, in the terminal, the error message is:
zsh: no such file or directory: ./configure

Help... Thank you

**Reply by ralfHielscher** [reply]
You need to execute the following lines in the directory of the NFFT:
```bash
./bootstrap.sh 
./configure --with-matlab=MATLAB_DIR --enable-nfsoft --enable-nfsft --enable-openmp --enable-portable-binary 
make
```

**Reply by ralfHielscher** [reply]
The point is that you forgot the `/bootstrap.sh` command which was part of step 1.

Ralf.

**Comment by ralfHielscher** [comment]
Hi Zach,

thank you for reporting this.

**Comment by Mary-alix** [comment]
Hello, 
Thank you very much for your quick replay. It is very helpful. 
but...

I downloaded manually nfft 3.5.2 from this page: https://www-user.tu-chemnitz.de/~potts/nfft/download/nfft-3.5.2.tar.gz
At the end of the installation the message error was: 
-configure: error: You don't seem to have the FFTW 3 library installed. You can #
     download it from http://www.fftw.org. If you have installed FFTW 3, #
     make sure that this configure script can find it. See ./configure #
     --help for more information.

I downloaded manually fftw-3.3.9, and I used the command: ./bootstrap.sh to install it. No error message at the end of the installation.

Than, I ran the command line you suggested: 
./bootstrap.sh 
./configure --with-matlab=MATLAB_DIR --enable-nfsoft --enable-nfsft --enable-openmp --enable-portable-binary 
make
-But I have the same error message with the FFTW 3 library not installed.

Do you have a suggestion for the next step I can do? ^-^
Thank you.
Mary-Alix

**Reply by ralfHielscher** [reply]
If you install the FFTW locally  you need to provide the configure script the path to the fftw binaries, i.e.,
```
./configure --with-fftw3-libdir=the_path_to_the_fftw_libs --all the other flags
```
Ralf.

**Reply by Mary-alix** [reply]
Dear Ralf, 

Thank you very much for your help.
I installed fftw 3.3.9 with brew to make sure I didn't make any mistake. Then I ran the ./configure command and still got the same error message.

Here is what the terminal tells me when I run the command:
checking whether directory /opt/homebrew/Cellar/fftw exists... yes
checking fftw3.h usability... no
checking fftw3.h presence... no
checking for fftw3.h... no
configure: error: You don't seem to have the FFTW 3 library installed. You can #
     download it from http://www.fftw.org. If you have installed FFTW 3, #
     make sure that this configure script can find it. See ./configure #
     --help for more information.

I don't know what to do. Do you have any suggestion (again)?
Thank you for your time.
Mary-Alix

**Reply by ralfHielscher** [reply]
Maybe you need to specify the path to libfftw3.so and fftw3.h separately with the options
`--with-fftw3-includedir=` and `--with-fftw3-libdir=`

Ralf.

**Reply by Mary-alix** [reply]
Hello again,

We added the two commands to the configure function and gave the local path to fftw3.h.
This worked well and the configure function found it.

However, the libfft3.so is not found on my computer. So we tried to input the local path to the library (/opt/homebrew/Cellar/fftw/3.3.9_1/lib). The new error message is:
checking fftw3.h usability... yes
checking fftw3.h presence... yes
checking for fftw3.h... yes
checking for library containing fftw_execute... no
configure: error: You don't seem to have the FFTW 3 library installed. You can #
     download it from http://www.fftw.org. If you have installed FFTW 3, #
     make sure that this configure script can find it. See ./configure #
     --help for more information.

We couldn't found this file: fftw_execute called in the error message.

Thank's for your patience, we really appreciate.

**Reply by michaelquellmalz** [reply]
When you installed fftw with Homebrew, it should usually be found automatically without specifying an additional path in the NFFT configure. You could easily check this by running `fftw-wisdom` from your command line (if the fftw is found, it prints the version number). 

Otherwise you have to set the path manually. It looks like the configure found the _fftw3.h_ file (in the directory of `--with-fftw3-includedir=`). But you need to specify the right ` --with-fftw3-libdir=` where the _libfftw3.dylib_ is. If it does not work with these configure flags, you could also set variables with the call
`CPPFLAGS=-I/fftwpath/include LDFLAGS=-L/fftwpath/lib ./configure --with-matlab=MATLAB_DIR --enable-nfsoft --enable-nfsft --enable-openmp`
where you replace `/fftwpath/include` by the path to fftw3.h and `/fftwpath/include` by the path to the dylib (Note to keep the letters -I and -L before the path).
